preprocess_era5.py

```diff
@@ -92,30 +92,27 @@ def load_data(variable, date):
         (cwv.sum(dim="level") - (w_upper + w_lower)).sum().values,
     )
 
-    # Change units to m3
-    # TODO: Check this! Change units before interp is tricky, if not wrong
-    total_seconds = config["timestep"] / config["divt"]
+    # Change units to m3, based on target frequency (not incoming frequency!)
+    target_freq = config['target_frequency']
+    total_seconds = pd.Timedelta(target_freq).total_seconds()
     fa_e_upper *= total_seconds * (l_ew_gridcell / density_water)
     fa_e_lower *= total_seconds * (l_ew_gridcell / density_water)
     fa_n_upper *= total_seconds * (l_mid_gridcell[None, :, None] / density_water)
     fa_n_lower *= total_seconds * (l_mid_gridcell[None, :, None] / density_water)
 
-    # Put data on a smaller time step...
-    time = w_upper.time.values
-    newtime = pd.date_range(time[0], time[-1], freq="15Min")[:-1]
-    w_upper = w_upper.interp(time=newtime).values
-    w_lower = w_lower.interp(time=newtime).values
+    # Increased time resolution; states at midpoints, fluxes at the edges
+    old_time = w_upper.time.values
+    newtime_states = pd.date_range(old_time[0], old_time[-1], freq=target_freq)
+    newtime_fluxes = newtime_states[:-1] + pd.Timedelta(target_freq) / 2
```

Comment:
Als je newtime_states als "newtime_states = pd.date_range(old_time[0], old_time[-1], freq=target_freq)[:-1]" veranderd, dan moet hier denk ik "[:-1]" verwijderd worden achter newtime_states.

Reply:
Dat is inderdaad veranderd, maar volgens mij is dit wel het gewenste gedrag.